File: Safari-609.1.11.1/LayoutTests/imported/blink/http/tests/xmlhttprequest/resources/redirect-cors-origin-null.php
<?php
    $request_origin_value = $_SERVER["HTTP_ORIGIN"];

    if ($_SERVER['REQUEST_METHOD'] == "GET") {
     header("HTTP/1.1 302");
     header("Location: http://localhost:8000/xmlhttprequest/resources/redirect-cors-origin-null-pass.php");
    }
    if (!is_null($request_origin_value)) {
        header("Access-Control-Allow-Origin: $request_origin_value");
    }
?>

File: LayoutTests/http/tests/xmlhttprequest/resources/redirect-cors-origin-null.py
#!/usr/bin/env python3
import os
import sys
import time

sys.stdout.write('Content-Type: text/html\r\n')

if os.environ.get('REQUEST_METHOD') == 'GET':
    sys.stdout.write(
        'status: 302\r\n'
        'Location: http://localhost:8000/xmlhttprequest/resources/redirect-cors-origin-null-pass.py\r\n'
    )

if os.environ.get('HTTP_ORIGIN'):
    sys.stdout.write('Access-Control-Allow-Origin: {}\r\n'.format(
        os.environ.get('HTTP_ORIGIN')
    ))

sys.stdout.write('\r\n')
